docs/cookbook/evals/evaluating_web_search_agent.md

```diff
@@ -157,11 +161,64 @@ class WebAssistant(BaseModel):
             print()
 ```
 
-For more information regarding this code snippet, refer to the Web Search Agent cookbook.
+## Basic Evaluations
+
+Let's start off with some basic evaluations to know whether our agent is working in general. Given that updates to prompts can significantly influence LLM behavior, it's crucial to test each component of our agent individually.
+
+### Evaluating `_web_search` tool
+
+Our goal is to ensure that the LLM consistently utilizes the web search tool, rather than relying on its inherent knowledge base to generate responses. We've intentionally refrained from explicitly instructing the agent to always utilize the web search tool, as some user queries may be more conversational in nature and not necessitate web searches. However, for user queries that are more information-seeking, the agent should always leverage the web search tool.
+
+```python
+import pytest
+
+user_queries = [
+    "How is the weather in New York?",
+    "What is the capital of France?",
+    "Who is the president of the United States?",
+    "What is the population of India?",
+    "What is an apple?",
+]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("user_query", user_queries)
+async def test_web_search(user_query: str):
+    """Tests that the web search agent always uses the web search tool."""
+    web_assistant = WebAssistant()
+    response = await web_assistant._stream(user_query)
+    async for _, tool in response:
+        assert tool._name() == "_web_search" if tool else False
+```
+
+It's recommended to continually expand our golden dataset until we can confidently assert that the LLM uses web search when appropriate.
+
+### Evaluating `extract_content` tool
+
+Our agent has been prompt engineered to utilize the extract_content tool at its discretion. Given the non-deterministic nature of this test, we'll implement a basic verification to ensure that the `extract_content` tool is invoked at least once per user query. We'll employ the same golden dataset used in the `test_web_search`, allowing us to assume that `test_extract_content` will always have a functional `_web_search`.
+
+```python
+@pytest.mark.asyncio
+@pytest.mark.parametrize("user_query", user_queries)
+async def test_extract_content(user_query: str):
```

Comment:
I wonder if we could somewhat simplify this test and make it more direct if we called `_step` with a history that has a previous tool call calling web_search? that way we can just assert that we get some number of extract_content calls, and we can test that extract_content is called with the correct link from the web search results?

Reply:
done